### PR TITLE
RSC SDK beta

### PR DESCRIPTION
**Pull request by steve8708**

### CODE REVIEW

**Review by samijaber — COMMENTED** [review]
@steve8708 can we delete all the redundant stuff in `e2e/rcs/src/*`? I assume it came from a starter and we don't need any of it.

**Comment by steve8708** [comment]
oh so that is actually incredibly useful right now for testing things thoroughly. if it's ok perhaps I can keep wrangling with that and then when things are looking good can distill it down and send a future PR with the most minimal e2e flow

**Comment by steve8708** [comment]
the amount of merge conflicts keeps growing, @samijaber can you give this a 👍 or 👎

**Comment by samijaber** [comment]
Oh sorry, I totally forgot to come back to this PR 🤦🏽

File: packages/sdks/src/blocks/util.ts [hunk 1]
```diff
@@ -1,6 +1,5 @@
-export function markSerializable<FN extends (...args: any[]) => any>(
```
Comment (samijaber) [inline packages/sdks/src/blocks/util.ts:1]: cc @mhevery I think you had some special logic here but stringifying the functions does the trick instead

File: packages/sdks/src/functions/get-block-styles.ts [hunk 2]
```diff
@@ -23,25 +23,8 @@ const getStyleForTarget = ({
...
-        ...(styles.small
-          ? {
-              [getMaxWidthQueryForSize('small')]: convertStyleObject(
-                styles.small
-              ),
-            }
-          : {}),
-      };
+      // By default we do not want inline styles
+      return {};
```
Comment (samijaber) [inline packages/sdks/src/functions/get-block-styles.ts:27]: I simplified all this logic recently to only run on React Native, so you can take my changes from the merge conflict.

File: packages/sdks/src/scripts/init-editing.ts [hunk 3]
```diff
@@ -23,7 +23,12 @@ export const registerInsertMenu = () => {
   });
 };
 
+let isSetupForEditing = false;
 export const setupBrowserForEditing = () => {
+  if (isSetupForEditing) {
+    return;
+  }
+  isSetupForEditing = true;
```
Comment (samijaber) [inline packages/sdks/src/scripts/init-editing.ts:31]: Why is this needed? Asking because these side-effects might not consistently work for all framework environments, so if preferable I'd rather not rely on them (or know exactly what they're needed for at least)

File: packages/sdks/src/blocks/util.ts [hunk 4]
```diff
@@ -1,6 +1,5 @@
...
+export function changeFunctionToString<FN extends (...args: any[]) => any>(
   fn: FN
-): FN {
-  (fn as any).__qwik_serializable__ = true;
-  return fn;
+): string {
+  return `return (${fn.toString()}).apply(this, arguments)`;
```
Comment (samijaber) [inline packages/sdks/src/blocks/util.ts:4]: note to self: this is the same as `serializeFn` in `src/functions/register-component.ts`, can dedupe.

**Review by samijaber — APPROVED** [review]

File: packages/sdks/src/blocks/util.ts (hunk 4 above)
Comment (samijaber) [reply packages/sdks/src/blocks/util.ts:4]: Actually, this function is simply making sure we serialize the fn ahead of time, `serializeFn` is no longer necessary

File: packages/sdks/src/scripts/init-editing.ts (hunk 3 above)
Comment (steve8708) [reply packages/sdks/src/scripts/init-editing.ts:31]: we could move to a window var - the point though is just that we want to be sure this only ever runs one time, not multiple, or we could face issues

Comment (samijaber) [reply packages/sdks/src/scripts/init-editing.ts:31]: `init-editing` runs in the `onMount` hook in every framework besides RSC though, so it's guaranteed to only run once per `RenderContent`. If this is RSC-specific, can probably add a piece of boolean state in `BuilderEditingWrapper` to make sure it only runs once.

Comment (steve8708) [reply packages/sdks/src/scripts/init-editing.ts:31]: that's exactly the problem, many people will have multiple <RenderContent /> components on one page, we don't want all this stuff to run multiple times (e.g. the message event binding for code execution - bad for perf and reliability to have these register multiple times)

Comment (steve8708) [reply packages/sdks/src/scripts/init-editing.ts:31]: also tmk this would not be an issue with bundlers, the issue with those is tree shaking, but the "side effect" is just a local var, I've never seen a case where this is problem with even the most aggressive tree shakers

Comment (samijaber) [reply packages/sdks/src/scripts/init-editing.ts:31]: Fair enough, we can go with this approach. 👍🏽
